homework/list.js

```diff
@@ -1,43 +1,53 @@
 'use strict';
 
-(function() {
+//the way of how to refactor list.js is the same as content.js 
+//repair some error in list.js
+// add promise into it
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
+(function(exports) {
 
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
+  var ListManager = function(){
+    this._listNoteContent = [];
+    this._wrapper = document.querySelector("#note-content-wrapper");
+  }
+
+  ListManager.prototype = {
+    start() {
+      this.fetchList().then((function(data) {
+      this.updateList(data);
+      this.drawList();
+      this.preloadFirstNote();
+    }).bind(this))
+    .catch((function () {
+    }).bind(this));
+    
+    window.addEventListener('click', (function(event) {
+      this.onNoteOpen(event).bind(this));
```

Comment:
It's a convention to prefix `on` to the method corresponds to the event name. So in fact it should be renamed as `onNoteClicked`.